rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py

```diff
@@ -0,0 +1,620 @@
+                                    self.on_waypoint =\
+                                        self.last_known_waypoint_index
+                                else:
+                                    self.on_lane = None  # update_off_grid()
+                                    self.on_waypoint = None
+                        # ------------------------ #
+                        # IMPLEMENT YOUR CODE HERE #
+                        # If your robot does not have an API to report the
+                        # remaining travel duration, replace the API call
+                        # below with an estimation
+                        # ------------------------ #
+                        duration = self.api.navigation_remaining_duration()
+                        if self.path_index is not None:
+                            self.next_arrival_estimator(
+                                self.path_index, timedelta(seconds=duration))
+            self.path_finished_callback()
+            self.node.get_logger().info(
+                f"Robot {self.name} has successfully navigated along "
+                f"requested path.")
+
+        self._follow_path_thread = threading.Thread(
+            target=_follow_path)
+        self._follow_path_thread.start()
+
+    def dock(
+            self,
+            dock_name,
+            docking_finished_callback):
+        ''' Docking is very specific to each application. Hence, the user will
+            need to customize this function accordingly. In this example, we
+            assume the dock_name is the same as the name of the waypoints that
+            the robot is trying to dock into. We then call api.start_process()
+            to initiate the robot specific process. This could be to start a
+            cleaning process or load/unload a cart for delivery.
+        '''
+
+        self._quit_dock_event.clear()
+        if self._dock_thread is not None:
+            self._dock_thread.join()
+
+        self.dock_name = dock_name
+        assert docking_finished_callback is not None
+        self.docking_finished_callback = docking_finished_callback
+
+        # Get the waypoint that the robot is trying to dock into
+        dock_waypoint = self.graph.find_waypoint(self.dock_name)
+        assert(dock_waypoint)
+        self.dock_waypoint_index = dock_waypoint.index
+
+        def _dock():
+            # Request the robot to start the relevant process
+            self.node.get_logger().info(
+                f"Requesting robot {self.name} to dock at {self.dock_name}")
+            self.api.start_process(self.dock_name, self.map_name)
+
+            with self._lock:
+                self.on_waypoint = None
+                self.on_lane = None
+            time.sleep(0.5)
+            # ------------------------ #
+            # IMPLEMENT YOUR CODE HERE #
+            # With whatever logic you need for docking #
+            # ------------------------ #
+            while (not self.api.docking_completed()):
```

Comment:
I think this is supposed to be `while (not self.api.process_completed()):`

Reply:
Yup you're right! Edited